src/icmp.rs

```diff
@@ -0,0 +1,353 @@
+//! Handles parsing of ICMP
+
+use nom::{IResult, Err, Needed, number};
+use crate::ipv4::{IPv4Header, parse_ipv4_header, address};
+use std::net::Ipv4Addr;
+
+#[derive(Clone, Copy, Debug, PartialEq, Eq)]
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+pub enum Unreachable {
+    DestinationNetworkUnreachable,
+    DestinationHostUnreachable,
+    DestinationProtocolUnreachable,
+    DestinationPortUnreachable,
+    FragmentationRequired,
+    SourceRouteFailed,
+    DestinationNetworkUnknown,
+    DestinationHostUnknown,
+    SourceHostIsolated,
+    NetworkAdministrativelyProhibited,
+    HostAdministrativelyProhibited,
+    NetworkUnreachableForTos,
+    HostUnreachableForTos,
+    CommunicationAdministrativelyProhibited,
+    HostPrecedenceViolation,
+    PrecedentCutoffInEffect,
+}
+
+#[derive(Clone, Copy, Debug, PartialEq, Eq)]
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+pub enum Redirect {
+    Network,
+    Host,
+    TosAndNetwork,
+    TosAndHost,
+}
+
+#[derive(Clone, Copy, Debug, PartialEq, Eq)]
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+pub enum TimeExceeded {
+    TTL,
+    FragmentReassembly
+}
+
+#[derive(Clone, Copy, Debug, PartialEq, Eq)]
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+pub enum ParameterProblem {
+    Pointer,
+    MissingRequiredOption,
+    BadLength
+}
+
+#[derive(Clone, Copy, Debug, PartialEq, Eq)]
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+pub enum ExtendedEchoReply {
+    NoError,
+    MalformedQuery,
+    NoSuchInterface,
+    NoSuchTableEntry,
+    MupltipleInterfacesStatisfyQuery
+}
+
+#[derive(Clone, Copy, Debug, PartialEq, Eq)]
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+pub enum IcmpCode {
+    EchoReply,
+    Reserved,
+    DestinationUnreachable(Unreachable),
+    SourceQuench,
+    Redirect(Redirect),
+    EchoRequest,
+    RouterAdvertisment,
+    RouterSolicication,
+    TimeExceeded(TimeExceeded),
+    ParameterProblem(ParameterProblem),
+    Timestamp,
+    TimestampReply,
+    ExtendedEchoRequest,
+    ExtendedEchoReply(ExtendedEchoReply),
+    Other(u16)
+}
+
+impl From<u16> for IcmpCode {
+    fn from(raw: u16) -> Self {
+        let bytes = raw.to_be_bytes();
+        let t = bytes[0];
+        let c = bytes[1];
```

Comment:
`let [t, c] = raw.to_be_bytes();`

Reply:
thanks,
fixed in https://github.com/bestouff/pktparse-rs/pull/12/commits/63056f68094a20de49f9d9b44600cf36b0401e86